ml_tools/eolearn/ml_tools/TDigestTask.py

```diff
@@ -0,0 +1,199 @@
+    It integrates the [T-Digest algorithm by Ted Dunning](https://arxiv.org/abs/1902.04023) to efficiently \
+        compute quantiles of the underlying dataset into eo-learn.
+    The output features of the tasks may be merged to compute a representation of the complete dataset.
+    That enables quantile based normalisation or statistical analysis of datasets larger than RAM in EO.
+    """
+
+    def __init__(self, in_feature, out_feature, *args, mode=None, pixelwise=True, **kwargs):
+        """
+        :param in_feature: The input feature to compute the T-Digest representation for.
+        :type in_feature: (FeatureType, str)
+        :param out_feature: The output feature where to save the T-Digest representation of the chosen feature.
+        :type out_feature: (FeatureType, str)
+        :param mode: The mode to apply to the timestamps and bands.
+        - The 'standard' mode computes the T-Digest representation \
+            for each band accumulating timestamps.
+        - The 'timewise' mode computes the T-Digest representation \
+            for each band and timestamp of the chosen feature.
+        - The 'monthly' mode computes the T-Digest representation \
+            for each band accumulating the timestamps per month.
+        - The 'weekly' mode computes the T-Digest representation \
+            for each band accumulating the timestamps per seven days.
+        - The 'daily' mode computes the T-Digest representation \
+            for each band accumulating the timestamps per day.
+        - The 'total' mode computes the total T-Digest representation \
+            of the whole feature accumulating all timestamps, bands and pixels.
+        :type mode: str
+        :param pixelwise: Decider whether to compute the T-Digest representation accumulating pixels or per pixel.
+        :type pixelwise: boolean
+        :param args: arguments for init of EOTask base class.
+        :type args: arguments
+        :param kwargs: keyword arguments for init of EOTask base class.
+        :type kwargs: keyword arguments
+        """
+        super().__init__(*args, **kwargs)
+
+        # check input and output features
+        if isinstance(in_feature, list):
+            self.in_feature = in_feature
+            if isinstance(out_feature, list):
+                assert len(in_feature) == len(out_feature)
+                self.out_feature = out_feature
+            else:
+                raise RuntimeError(
+                    "TDigestTask: you cannot give a list of input features without specifying the a list of output"
+                    " features of the same length!"
+                )
+        else:
+            self.in_feature = [in_feature]
+            if isinstance(out_feature, list):
+                raise RuntimeError(
+                    "TDigestTask: you cannot get a list of T-Digest representations from a single feature!"
+                )
+            else:
+                self.out_feature = [out_feature]
```

Comment:
If i understand correctly, here you have either 1 feature mapping to 1 feature, or N features mapping to N features?
In any case, i suggest you use our *feature parsers* which also support a large range of inputs. The easiest way for you is to take a look at [the init of the MapFeatureTask](https://github.com/sentinel-hub/eo-learn/blob/master/core/eolearn/core/core_tasks.py#L404) which has a similar approach when it comes to features.

This way you also don't need to think about 'do I have a single feature, do i have many'. If a single feature is passed as input, it gets transformed into a list with a single element, and you should be able to work with it as if it were a regular list of features.